app/login/layout.tsx

```diff
@@ -0,0 +1,24 @@
+import { decodeToken } from "@/lib/user";
+import { redirect } from "next/navigation";
+import { cookies } from "next/headers";
+
+export const metadata = {
+  title: "Login | Ovoflow"
+}
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  const cookieStore = cookies();
+  const token = cookieStore.get("token")?.value;
+
+  if (token) {
```

Comment:
Do we want to convert this to a `isLoggedIn()` helper function?

Reply:
Implemented function `getAuthRole()` to get the authorized role from the token